src/acceleration/impl/ParallelMatrixOperations.hpp

```diff
@@ -280,28 +280,28 @@ class ParallelMatrixOperations {
 
     // sum up blocks in master, reduce
     Eigen::MatrixXd summarizedBlocks = Eigen::MatrixXd::Zero(p, r); /// @todo: only master should allocate memory.
-    utils::MasterSlave::reduceSum(block, summarizedBlocks);
+    utils::IntraComm::reduceSum(block, summarizedBlocks);
 
     // slaves wait to receive their local result
-    if (utils::MasterSlave::isSlave()) {
+    if (utils::IntraComm::isSecondary()) {
       if (result.size() > 0)
-        utils::MasterSlave::getCommunication()->receive(result, 0);
+        utils::IntraComm::getCommunication()->receive(result, 0);
     }
 
     // master distributes the sub blocks of the results
-    if (utils::MasterSlave::isMaster()) {
+    if (utils::IntraComm::isPrimary()) {
       // distribute blocks of summarizedBlocks (result of multiplication) to corresponding slaves
       result = summarizedBlocks.block(0, 0, offsets[1], r);
 
-      for (Rank rankSlave : utils::MasterSlave::allSlaves()) {
-        int off       = offsets[rankSlave];
-        int send_rows = offsets[rankSlave + 1] - offsets[rankSlave];
+      for (Rank rankSecondary : utils::IntraComm::allSecondaries()) {
+        int off       = offsets[rankSecondary];
+        int send_rows = offsets[rankSecondary + 1] - offsets[rankSecondary];
```

Comment:
I'm not sure how many times this occurs, but we should maybe rename it to `secondaryRank`

```suggestion
      for (Rank secondaryRank : utils::IntraComm::allSecondaries()) {
        int off       = offsets[secondaryRank];
        int send_rows = offsets[secondaryRank + 1] - offsets[secondaryRank];
```